.github/workflows/automation-scripts/update-changelogs.mjs

```diff
@@ -53,9 +50,14 @@ await $`git checkout ${targetBranch}`;
 // copy all changelogs from the target branch to ./temp-target-changelogs, the files will be named: package_name_CHANGELOG.json
 await $`find ./ -type f -name "CHANGELOG.json" -not -path "*/node_modules/*" -exec sh -c 'cp "{}" "./temp-target-changelogs/$(echo "{}" | sed "s/^.\\///; s/\\//_/g")"' \\;`;
 
-const currentFiles = getFilePaths(targetPath);
+const currentFiles = fs.readdirSync(targetPath);
+const incomingFiles = fs.readdirSync(incomingPath);
+// Do not include packages from Current branch if they do not exist in the Incoming branch, ie. new packages in later versions of itwinjs-core.
 currentFiles.forEach((file, index) => {
```

Comment:
I think either of these implementations might be easier to understand. I slightly prefer the for loop just cause I don't think that helper function is too necessary, but I'm fine with either

`const targetFiles = allTargetFiles.filter(excludeNewPkgs)
function excludeNewPkgs(file) {
  if (incomingFiles.includes(file))
    return file
  else
    console.log(`${file} is not a package in ${currentBranch}. Skipping this package.`);
}

const targetFiles = [];
allTargetFiles.forEach((file) => {
  if (incomingFiles.includes(file))
    targetFiles.push(file);
  else
    console.log(`${file} is not a package in ${currentBranch}. Skipping this package.`);
})
`

Reply:
I think you can just inline your helper function for the filter part, similar to the forEach.
```
const targetFiles = allTargetFiles.filter((file) => {
  if (incomingFiles.includes(file)) {
      return file;
  }
  else {
      console.log(`${file} is not a package in ${currentBranch}. Skipping this package.`);
  }
})
```

Reply:
Yeah, I think that that either of the solutions you have laid out are much easier to understand than the foreach and splice solution that you currently have. Like An mentioned, you don't have to the define the helper function like that, you can just inline it in the filter call. I personally prefer the filter solution, since the array can just be instantiated to the return value instead of pushing values in each iteration of a loop.

Reply:
agreed, thanks guys